content/posts/react-in-python-with-pyodide.md

```diff
@@ -0,0 +1,435 @@
+
+# Let's use everything created above in our App
+@pythonify
+def App(props, children):
+    return Container(...).update(
+        div(
+            # ...
+            ProTip(),
+            Copyright(),
+            style=dict(...),
+        ),
+    )
+```
+
+> The full demo can be found in `demos/react-in-pyodide/demo-5` or [on JSFiddle](https://jsfiddle.net/gh/get/library/pure/pyodide/pyodide-blog/contents/demos/react-in-pyodide/demo-5/).
+
+
+## Separating HTML and Python
+
+If you prefer to write your python code inside `.py` files, you can simply move all of the Python code to a separate `main.py` file and fetch it from your HTML file:
+
+```html
+<body>
+    <script type="text/javascript">
+        async function main() {
+            let pyodide = await loadPyodide({
+                indexURL: "https://cdn.jsdelivr.net/pyodide/v0.18.0/full/"
+            });
+
+            const script = await fetch("/path/to/main.py");
+            const scriptText = await script.text();
+            pyodide.runPython(scriptText);
+        }
+
+        main();
+    </script>
+</body>
+```
+
+Note that the absolute path needs to be specified, and will differ based on your provider. [This repo](https://github.com/xhlulu/material-ui-in-pyodide) shows you how it works on Github Pages. In general, JS will try to look for `yourwebsite.com/path/to/main.py` and will be able to fetch it if a text file exists at that location.
+
+
+## Conclusion
+
+In summary, this post covered how to use Python to implement fundamental React concepts (such as hooks) and simple MUI components, and proposed the use of a `pythonify` function to make the component creation process feel more natural. 
+
+Moving forward, it will be interesting to see how we can leverage Pyodide to build React UIs using the thousands of React libraries available on [npm](https://www.npmjs.com/search?q=keywords:react) along with pure Python libraries on PyPi. Rather than replacing other Python UI frameworks, React+Pyodide would instead complement them. For example, whereas Dash allows you to write Python code on the server side (e.g. to process GBs of data and running GPU computations), it still requires you to use JS to create [custom components](https://github.com/plotly/dash-component-boilerplate) and [client-side callbacks](https://dash.plotly.com/clientside-callbacks). To fill this gap, React+Pyodide would allow you to create sophisticated UIs on the client side (e.g. to render a large number of components) without leaving Python, so community effort to connect those two frameworks could allow a full end-to-end workflow for building `pydata` apps in Python on both server and client-side.
+
+
+> **About this author:** *Xing Han Lu is a graduate student at Mila, where he works on NLP research, and previously worked as a ML developer at Plotly. He authored ML libraries on GitHub and publish code and data on Kaggle, where he is a Code Grandmaster.*
```

Comment:
Thanks the rendering looks good! Do you want to also link to your personal website and or mention your twitter username?

```suggestion
> **About the author** 
>
> *[Xing Han Lu](https://xinghanlu.com/) ([@xhluu](https://twitter.com/xhluu)) is a graduate student at Mila, where he works on NLP research, and previously worked as a ML developer at Plotly. He authored ML libraries on GitHub and publish code and data on Kaggle, where he is a Code Grandmaster.*
```